stl/inc/utility

```diff
@@ -644,6 +645,82 @@ template <size_t _Idx>
 inline constexpr in_place_index_t<_Idx> in_place_index{};
 #endif // _HAS_CXX17
 
+#if _HAS_CXX20
+
+template <class _Ty>
+inline constexpr bool _Is_integral_not_bool_char = is_integral_v<_Ty>
+    && !_Is_any_of_v<_Ty, bool, char,
+#ifdef _NATIVE_WCHAR_T_DEFINED
+        wchar_t,
+#endif // _NATIVE_WCHAR_T_DEFINED
+#ifdef __cpp_char8_t
+        char8_t,
+#endif // __cpp_char8_t
+        char16_t, char32_t>;
+
+// SAFE INTEGRAL COMPARISONS
+template <class _Ty1, class _Ty2>
+constexpr bool in_range(const _Ty2 _Value) noexcept {
+    static_assert(_Is_integral_not_bool_char<_Ty1> && _Is_integral_not_bool_char<_Ty2>);
+    if constexpr (is_signed_v<_Ty1> && is_unsigned_v<_Ty2>) {

```

Comment:
I think I'd rather see `!is_signed_v` than `is_unsigned_v` in all of these functions - there's no need to instantiate four variable templates when two would do. (Equivalently, you could reorder your cases as the working draft does:
```c++
if constexpr (both are signed) {}
else if constexpr (left is signed) {}
else {}
```
which I would mildly prefer, but that's simply style preference so your call.)